airflow/cli/commands/remote_commands/config_command.py

```diff
@@ -546,6 +560,9 @@ def lint_config(args) -> None:
         ):
             lint_issues.append(configuration.message)
 
+        if not args.skip_problematic_default_checks:
+            conf.handle_incompatible_airflow2_defaults(upgrade=args.upgrade_problematic_defaults)
+
```

Comment:
Which means, since this has already run and upgraded, you can get rid of this completely - and the args, etc. All of it.

Reply:
It is unfortunate that this is what you get:

```
$ airflow config lint
/Users/jedc/github/airflow/airflow/configuration.py:440 FutureWarning: [core] dag_ignore_file_syntax: The default value changed from 'regexp' in Airflow 2.x to 'glob' in Airflow 3.0. Auto-upgraded from 'regexp' to 'glob'.
No issues found in your airflow.cfg. It is ready for Airflow 3!
```

The warning is from `configuration.py`, where it got auto-upgraded.

Reply:
This was a behaviour that I intentionally added where it automatically upgraded by default. If we want to change this to upgrade only via `airflow config lint`, I can pass `upgrade=False` in configuration.py method.